### PR TITLE
Add Juniper EX4100 device types modified from EX3400 models

### PR DESCRIPTION
**Pull request by twilley**
These are based on the EX3400 device types.

### CODE REVIEW

**Comment by twilley** [comment]
@danner26 I'm modeling that now, and I will commit soon.  Is there a solution I'm not aware of to provide the preinstallation of modules within the module bays when the device is created? I would love to see a way to provide pre-installed modules in a module bay (or devices in a device bay). For example, the EX4100-48P comes preinstalled with two fan modules and one 920w power supply. When deploying many (or even just a handful) of these devices in Netbox, it seems like quite a burden to have to create and install all of the fan modules and power supplies. I suppose the device import can accomplish this, but I've not used it yet.

**Comment by danner26** [comment]
Hello, currently there is no way to pre-define the installation of modules. You can automate that using something like ansible, the API or doing bulk imports, but otherwise there is not currently a way.

The API and subsequent programatic methods of interacting with NetBox will be your best friend when importing a lot of data into the system

File: device-types/Juniper/EX4100-48T.yaml [hunk 1]
```diff
@@ -0,0 +1,135 @@
+---
+manufacturer: Juniper
+model: EX4100-48T
+slug: ex4100-48t
```
Comment (danner26) [inline device-types/Juniper/EX4100-48T.yaml:4]: Slugs should have the manufacturer appended to them. For this instance, `juniper-`

File: device-types/Juniper/EX4100-48T.yaml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+u_height: 1
```
Comment (danner26) [inline device-types/Juniper/EX4100-48T.yaml:5]: Please add the following missing values, if you have them:

1. `part_number`
2. `is_full_depth`
3. `airflow`
4. `weight` and `weight_unit`

If you do not have, or cannot find any of these values just let me know. Often you canfind all of these on the manufacturers datasheet for this product

File: device-types/Juniper/EX4100-48P.yaml [hunk 3]
```diff
@@ -0,0 +1,135 @@
+---
+manufacturer: Juniper
+model: EX4100-48P
+slug: ex4100-48p
+u_height: 1
```
Comment (danner26) [inline device-types/Juniper/EX4100-48P.yaml:5]: Please add the following missing values, if you have them:

1. `part_number`
2. `is_full_depth`
3. `airflow`
4. `weight` and `weight_unit`

If you do not have, or cannot find any of these values just let me know. Often you canfind all of these on the manufacturers datasheet for this product

File: device-types/Juniper/EX4100-24T.yaml [hunk 4]
```diff
@@ -0,0 +1,87 @@
+---
+manufacturer: Juniper
+model: EX4100-24T
+slug: ex4100-24t
+u_height: 1
```
Comment (danner26) [inline device-types/Juniper/EX4100-24T.yaml:5]: Please add the following missing values, if you have them:

1. `part_number`
2. `is_full_depth`
3. `airflow`
4. `weight` and `weight_unit`

If you do not have, or cannot find any of these values just let me know. Often you canfind all of these on the manufacturers datasheet for this product

File: device-types/Juniper/EX4100-24P.yaml [hunk 5]
```diff
@@ -0,0 +1,87 @@
+---
+manufacturer: Juniper
+model: EX4100-24P
+slug: ex4100-24p
+u_height: 1
```
Comment (danner26) [inline device-types/Juniper/EX4100-24P.yaml:5]: Please add the following missing values, if you have them:

1. `part_number`
2. `is_full_depth`
3. `airflow`
4. `weight` and `weight_unit`

If you do not have, or cannot find any of these values just let me know. Often you canfind all of these on the manufacturers datasheet for this product

**Review by danner26 — CHANGES_REQUESTED** [review]

**Review by danner26 — APPROVED** [review]

**Pull request merged**